src/base/detail/InitializedValue.hh

```diff
@@ -0,0 +1,57 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file InitializedValue.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Clear the value of the object on initialization and moving.
```

Comment:
What exactly is special about moving here?  If you wrote
```cpp
struct 
{
    int x = 0;
    double y = 10;
} foo;

do_something(std::move(foo));
```
Wouldn't the moved struct retain the values at initialization of the object, unless they were explicitly changed, in which case the moved object would have those values?

Reply:
Did you answer here?  Or was my question just stupid 😄

Reply:
Arggg I left a long response, I must have left the page without sending it.

Basically, `std::move` and `&&` references don't change the data (remember, `std::move` is only a cast that allows move functions to be invoked). Data only changes if someone implements a move/copy constructor that modifies the incoming data. So no, the moved object doesn't get its data reset automatically: the default move constructor for trivially copyable objects is just a copy constructor.

Reply:
I think I was more referring to the comment for the class:
```
Clear the value of the object on initialization and moving.
```
Since the data isn't cleared on a move. I guess the value of the class is that you get benefit of explicit initialization without writing initializer lists or adding inline initialization to members.